lib/install/management/finder.go

```diff
@@ -281,22 +280,55 @@ func (d *Dispatcher) searchVCHsPerDC(dc *object.Datacenter) ([]*vm.VirtualMachin
 	d.session.Finder.SetDatacenter(dc)
 
 	var vchs []*vm.VirtualMachine
-	if pools, err = d.session.Finder.ResourcePoolList(d.op, "*"); err != nil {
-		if _, ok := err.(*find.NotFoundError); ok {
-			return vchs, nil
-		}
+
+	pools, err = d.listResourcePools("*")
+	if err != nil {
 		err = errors.Errorf("Failed to search resource pools for datacenter %q: %s", dc.InventoryPath, err)
 		return nil, err
 	}
 
+	vchs = d.searchVCHsFromPools(pools)
+	return vchs, nil
+}
+
+// listResourcePool outputs a list of all resource pools under a compute path
+// retries on ObjectNotFound error because sometimes it's due to concurrent modifications of resource pools
+func (d *Dispatcher) listResourcePools(path string) ([]*object.ResourcePool, error) {
+	var err error
+	var pools []*object.ResourcePool
+
+	pools, err = d.session.Finder.ResourcePoolList(d.op, path)
+
+	// under some circumstances, such as when there is concurrent vic-machine delete operation running in the background,
+	// listing resource pools might fail because some VCH pool is being destroyed at the same time.
+	// If that happens, we retry and list pools again
+	err = retry.Do(func() error {
+		pools, err = d.session.Finder.ResourcePoolList(d.op, path)
+		if _, ok := err.(*find.NotFoundError); ok {
+			return nil
+		}
+		return err
+	}, func(err error) bool {
+		return tasks.IsTransientError(d.op, err) || tasks.IsNotFoundError(err)
+	})
+
+	return pools, err
+}
+
+// searchVCHsFromPools outputs all VCH VMs under the pools specified in the list in the argument.
+func (d *Dispatcher) searchVCHsFromPools(pools []*object.ResourcePool) []*vm.VirtualMachine {
+	var vchs []*vm.VirtualMachine
 	for _, pool := range pools {
 		children, err := d.getChildVCHs(pool, true)
+		// #nosec: Errors unhandled.
```

Comment:
Does it really give a warning here so you use "#nosec"? You explicitly checking for an error.

Reply:
No...but still I want to note it out that the error is simply logged but not "handled"
I guess it's not the right usage of #nosec

Reply:
#nosec is a tag for 'gas' tool, not humans.